### PR TITLE
Add assertions to BM cluster controller test

### PR DESCRIPTION
**Pull request by kashifest**
This PR makes small improvement of the BM cluster controller test by adding assertions on the type of error and re-queue duration.

### CODE REVIEW

File: controllers/baremetalcluster_controller_test.go [hunk 1]
```diff
@@ -56,11 +60,16 @@ var _ = Describe("Reconcile Baremetalcluster", func() {
 			res, err := r.Reconcile(req)
 			if tc.ErrorExpected {
 				Expect(err).To(HaveOccurred())
+				if tc.ErrorType != nil {
+					Expect(reflect.TypeOf(tc.ErrorType) == reflect.TypeOf(errors.Cause(err))).To(BeTrue())
```
Comment (Xenwar) [inline controllers/baremetalcluster_controller_test.go]: The last .To(BeTrue()) seems redundant.

Comment (kashifest) [reply controllers/baremetalcluster_controller_test.go]: Changed

**Pull request merged**